cdn_lambda/functions/origin_response/origin_response.py

```diff
@@ -24,6 +25,22 @@ def handler(self, event, context):
         request = event["Records"][0]["cf"]["request"]
         response = event["Records"][0]["cf"]["response"]
 
+        try:
+            want_digest = request["headers"]["want-digest"][0]["value"]
+        except KeyError:
+            LOG.debug(
+                "Could not read want-digest from response", exc_info=True
+            )
+            want_digest = None
+
+        if want_digest == "sha-256":
```

Comment:
I think this is not the right strategy for matching want_digest header... the caller is allowed to request multiple digest types (e.g. if they support multiple) and this would break when the caller does that.

There was an example on https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Want-Digest

> Want-Digest: SHA-512;q=0.3, sha-256;q=1, md5;q=0

I don't think we should silently ignore the Want-Digest request in that case.

Actually, it is legal for the server to respond with *any* digest type if the client asked for an unsupported type.  I think that means we don't need to look at the content of the header at all.  We can simply do "if the caller provided any Want-Digest header, give sha-256 digest".